leshan-lwm2m-core/src/main/java/org/eclipse/leshan/core/response/ReadCompositeResponse.java

```diff
@@ -16,19 +16,58 @@
 package org.eclipse.leshan.core.response;
 
 import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import org.eclipse.leshan.core.ResponseCode;
 import org.eclipse.leshan.core.node.LwM2mNode;
 import org.eclipse.leshan.core.node.LwM2mPath;
+import org.eclipse.leshan.core.node.TimestampedLwM2mNodes;
 
 public class ReadCompositeResponse extends AbstractLwM2mResponse {
 
     protected final Map<LwM2mPath, LwM2mNode> content;
 
-    public ReadCompositeResponse(ResponseCode code, Map<LwM2mPath, LwM2mNode> content, String errorMessage,
-            Object coapResponse) {
+    protected final TimestampedLwM2mNodes timestampedValues;
+
+    public ReadCompositeResponse(ResponseCode code, Map<LwM2mPath, LwM2mNode> content,
+            TimestampedLwM2mNodes timestampedValues, String errorMessage, Object coapResponse) {
         super(code, errorMessage, coapResponse);
-        this.content = content;
+
+        Map<LwM2mPath, LwM2mNode> responseContent;
+        TimestampedLwM2mNodes responsetimestampedValues;
+
+        if (timestampedValues != null) {
+            // handle if timestamped value is passed
+            if (content != null) {
+                throw new IllegalArgumentException("content OR timestampedValue should be passed but not both");
+            }
+            // store value if all timestamps in timestampedValues are null
+            if (!timestampedValues.getNodes().isEmpty()
+                    && timestampedValues.getTimestamps().stream().noneMatch(Objects::nonNull)) {
```

Comment:
Supposing we store timestamp as explain above,

1. `getNodes()` do a kind of merge through all timestamp I'm not sure this is really what you want ? (In fact as we expect only 1 timestamp the merge will do want we want but it will create a new map for not so much), So maybe better to do `timestampedValues.getNodesAt(timestamp);`

2. maybe I'm a bit all school but this code sounds a bit complicated... is it more simple to write it  like : 
`if (timestamp == null)`

Reply:
OK